rpcd/playbooks/roles/elasticsearch/tasks/upgrade_pre_flight.yml

```diff
@@ -0,0 +1,35 @@
+---
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Install Pre Flight Checks
+  template:
+    src: upgrade-pre-flight-checks.py.j2
+    dest: /opt/upgrade-pre-flight-checks.py
+    owner: root
+    group: root
+    mode: 0755
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - es-pre-flight
+
+- name: Run Pre Flight Checks
+  shell: /opt/upgrade-pre-flight-checks.py
```

Comment:
Right, so the pre-flight-check returns 1 if ES is already at 2.x and 2 if it's in the 1.7.x temporary state.  Ideally we would just skip all of the upgrade stuff if the rc is 1 and bail if the rc is 2.  If the rc is 0 we continue with the upgrade.